src/Ship.cpp

```diff
@@ -1135,24 +1150,34 @@ void Ship::UpdateAlertState()
 			SetAlertState(ALERT_SHIP_FIRING);
 			changed = true;
 		}
+		if (missile_detected) {
+			m_lastFiringAlert = Pi::game->GetTime();
+			SetAlertState(ALERT_MISSILE_DETECTED);
+			changed = true;
+		}
 		break;
 
 	case ALERT_SHIP_NEARBY:
-		if (!ship_is_near) {
+		if (!ship_is_near && !missile_detected) {
 			SetAlertState(ALERT_NONE);
 			changed = true;
 		} else if (ship_is_firing) {
 			m_lastFiringAlert = Pi::game->GetTime();
 			SetAlertState(ALERT_SHIP_FIRING);
 			changed = true;
+		} else if (missile_detected) {
+			m_lastFiringAlert = Pi::game->GetTime();
+			SetAlertState(ALERT_MISSILE_DETECTED);
+			changed = true;
 		}
 		break;
 
 	case ALERT_SHIP_FIRING:
-		if (!ship_is_near) {
+	case ALERT_MISSILE_DETECTED:
```

Comment:
Please assume I'm talking out of my ass, but just make sure this is correct, and it shouldn't be 
```
case ALERT_SHIP_FIRING || ALERT_MISSILE_DETECTED:
```
Since I can't, at the moment, understand what happens with ship firing alert state, if it's normal pew-pew combat without missiles firing?

maybe @Web-eWorks  or @fluffyfreak  or other cpp-ers can have a look at this commit.

Reply:
It should work as before

Reply:
@impaktor this is C-style `switch` so you don't get arbitrary expressions in your case statements. Fallthrough (like in this commit) is the only way to do that.